Newspack/NewspackShare/Controllers/ShareMediaViewController.swift

```diff
@@ -0,0 +1,331 @@
+            // This should never happen.
+            fatalError()
+        }
+        return ShareExtractor(extensionContext: self.extensionContext!, tempDirectory: tmpDir)
+    }()
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        coordinator.animate(alongsideTransition: { (context) in
+            self.collectionView?.reloadData()
+        }, completion: nil)
+
+        super.viewWillTransition(to: size, with: coordinator)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureStyle()
+        configureNav()
+        configureShadows()
+
+        guard validateData() else {
+            interruptSharing()
+            return
+        }
+        // TODO: Check that we have valid data to work with.
+        // There should be shadow sites, and at least one target story.
+        // If missing show error.
+
+
+        configureSharedItems()
+    }
+
+    func configureStyle() {
+        ShareAppearance.style(view: view, tableView: tableView)
+    }
+
+    func configureNav() {
+        navigationItem.title = NSLocalizedString("Share", comment: "Verb. Title of the screen shown when sharing from another app to Newspack.")
+    }
+
+    func configureShadows() {
+        shadowSites = shadowManager.retrieveShadowSites()
+
+        guard
+            let currentSiteIdentifer = UserDefaults.shared.string(forKey: AppConstants.currentSiteIDKey),
+            let currentStoryIdentifier = UserDefaults.shared.string(forKey: AppConstants.lastSelectedStoryFolderKey + currentSiteIdentifer)
+        else {
+            return
+        }
+        for site in shadowSites {
+            for story in site.stories {
+                if story.uuid == currentStoryIdentifier {
+                    targetStory = story
+                    return
+                }
+            }
+        }
+    }
+
+    func configureSharedItems() {
+        extracter.loadShare { (extracted) in
+            self.handleSharedItems(items: extracted)
+        }
+    }
+
+}
+
+// MARK: - Validation and prompts
+
+extension ShareMediaViewController {
+
+    /// In order to share to the main app, the main app must have at least one
+    /// site and at least one story. Since we don't have direct access to the main
+    /// app's data, we'll trust that shadowSites are an accurate reflection. If
+    /// either shadowSites are empty, or there is not a target story retrieved
+    /// we'll assume the app has no sites or stories.
+    /// Note that there is an edge case where shadowSites and a targetStory could
+    /// be stale, and the app actually have no sites or stories (e.g. if the site
+    /// or story folders were manually deleted via the files app.
+    ///
+    func validateData() -> Bool {
+        guard
+            shadowSites.count > 0,
+            let _ = targetStory
+        else {
+            return false
+        }
+
+        return true
```

Comment:
Nipticky!! (Feel super free to disregard!!)

```
return shadowSites.count > 0 && targetStory != nil
```

Reply:
:)  I like the longer form for readability at a glance.   Good suggestion tho! Keep 'em coming!

Reply:
Hmm... Actually after a quick second thought... I like you're suggestion better!  Call me fickle 😆